File: Projectile.py
import math
import numpy as np
import matplotlib.pyplot as plt

g = 9.81

def projectile_motion(initial_velocity, angle, t, time_interval):
    angle = math.radians(angle)

    Ux = initial_velocity * math.cos(angle)
    Uy = initial_velocity * math.sin(angle) - g * t

    xt = Ux * t
    yt = (initial_velocity * math.sin(angle)) * t - 0.5 * g * t**2

    maxHeight = (initial_velocity * math.sin(angle))**2 / (2 * g)

    Range = (initial_velocity**2 * math.sin(2 * angle)) / g

    time_of_flight = 2 * (initial_velocity * math.sin(angle)) / g

    print(f"Horizontal component of velocity (Ux): {Ux:.2f} m/s")
    print(f"Vertical component of velocity (Uy): {Uy:.2f} m/s")
    print(f"Horizontal displacement at time t: {xt:.2f} m")
    print(f"Vertical displacement at time t: {yt:.2f} m")
    print(f"Time of flight: {time_of_flight:.2f} s")
    print(f"Maximum height: {maxHeight:.2f} m")
    print(f"Horizontal range: {Range:.2f} m")

    t_values = np.arange(0, t + time_interval, time_interval)
    y_values = []
    for i in t_values:
        xt = Ux * i
        y = math.tan(angle)* xt - (g * (xt ** 2)) / (2 * (Ux ** 2))
        y_values.append(y)
        print(f"At t = {i:.2f}, y = {y:.2f} m")

    plt.plot(t_values, y_values)
    plt.xlabel("Time (s)")
    plt.ylabel("Height (m)")
    plt.title("Projectile Motion")
    plt.grid(True)
    plt.show()

initial_velocity = float(input("Enter the initial velocity (m/s): "))
angle = float(input("Enter the angle of projection (in degrees): "))
t = float(input("Enter the time of projection (in seconds): "))
time_interval = float(input("Enter the time interval (in seconds): "))

projectile_motion(initial_velocity, angle, t, time_interval)

File: Projectile.c
#include <stdio.h>
#include <math.h>

#define g 9.81 

void projectileMotion(float initial_velocity, float angle, float t, float time_interval) {
    float Ux, Uy, xt, yt, maxHeight, range, time_of_flight, y;

    angle = angle * M_PI / 180.0;

    Ux = initial_velocity * cos(angle);
    Uy = initial_velocity * sin(angle) - g * t;

    xt = Ux * t;
    yt = (initial_velocity * sin(angle)) * t - 0.5 * g * pow(t, 2);

    maxHeight = (pow(initial_velocity * sin(angle), 2)) / (2 * g);

    range = (pow(initial_velocity, 2) * sin(2 * angle)) / g;

    time_of_flight = 2 * (initial_velocity * sin(angle)) / g;

    printf("Horizontal component of velocity (Ux): %.2f m/s\n", Ux);
    printf("Vertical component of velocity (Uy): %.2f m/s\n", Uy);
    printf("Horizontal displacement at time t: %.2f m\n", xt);
    printf("Vertical displacement at time t: %.2f m\n", yt);
    printf("Time of flight: %.2f s\n", time_of_flight);
    printf("Maximum height: %.2f m\n", maxHeight);
    printf("Horizontal range: %.2f m\n", range);
    
    for (float i = 0; i <= t; i += time_interval) {
        xt = Ux * i;
        y = tan(angle) * xt - (g * pow(xt, 2)) / (2 * pow(Ux, 2));
        printf("At t = %.2f, y = %.2f m\n", i, y);
    }
}

int main() {
    float initial_velocity, angle, t, time_interval;

    printf("Enter the initial velocity (m/s): ");
    scanf("%f", &initial_velocity);

    printf("Enter the angle of projection (in degrees): ");
    scanf("%f", &angle);

    printf("Enter the time of projection (in seconds): ");
    scanf("%f", &t);

    printf("Enter the time interval (in seconds): ");
    scanf("%f", &time_interval);

    projectileMotion(initial_velocity, angle, t, time_interval);

    return 0;
}